src/metabase/events/view_log.clj

```diff
@@ -90,6 +90,33 @@
       (catch Throwable e
         (log/warnf e "Failed to process view event. %s" topic)))))
 
+(derive ::dashboard-queried :metabase/event)
+(derive :event/dashboard-queried ::dashboard-queried)
+
+(def ^:private update-dashboard-last-viewed-at-interval-seconds 20)
+
+(defn- update-dashboard-last-viewed-at!* [dashboard-ids]
+  (t2/update! :model/Dashboard {:id [:in (into #{} dashboard-ids)]} {:last_viewed_at :%now}))
```

Comment:
Setting all these dashboards to have `last_used_at = now` is inaccurate because `now` is the batch processing time.

We can be more accurate here if you record the `submitted` time of a dashboard and use a case statement to update it.

I did a similar thing in https://github.com/metabase/metabase/blob/1a07aa45ed7f1e9f1b7b137672855b8d578b73a1/src/metabase/query_processor/middleware/update_used_cards.clj#L23

Reply:
I thought about this - are you sure it's worth the additional complexity to be precise here?

Maybe we should change the name of the field to `:not_used_after`/`:not_viewed_after`, but I think that we can not worry too much about being ~20s later than reality. The times are best effort anyway - ideally we'll err on the later side, but even that isn't always possible:
- the current `:last_viewed_at` may be whatever time the migration was run to create this column
- it may be one of the times the card/dashboard was used, but some subsequent usages were lost

It seems to me that the times here are approximate and that's ok - as far as I know the only planned use of this column is the "cleanup stale items" project, and if we don't cleanup something that was used 6 months minus 20 seconds ago when we should have, I don't think anyone will notice or complain 😅

Reply:
hmm, sorry but I don't see the downside of doing that, it's quite simple IMO.

Also we currently set the batch process time to 20s but there is nothing stopping us from increasing that in the future.

Reply:
Ok, that makes sense - done